### PR TITLE
summary.py: make more robust against garbage input

### PR DESCRIPTION
**Pull request by lsf37**
In some scenarios Bamboo might deliver corrupted, partial, empty, or
missing property files. These changes guard against that scenario to
not fail the build in these cases and still produce some output that
might be usable.

Signed-off-by: Gerwin Klein <kleing@unsw.edu.au>

### CODE REVIEW

File: bamboo/summary.py [hunk 1]
```diff
@@ -92,8 +104,9 @@ def pr(line):
             pr("<table><tbody>")
             first = True
             for job in failed:
+                job_name = job_names[job] if job in job_names else job
```
Comment (oliver-wm) [inline bamboo/summary.py:107]: job_name = job_names.get(job, job)

File: bamboo/summary.py [hunk 2]
```diff
@@ -70,8 +77,13 @@ def pr(line):
             pr("<table>")
             first = True
             for repo in repos:
-                pr('<tr {4}> <td class="col1"><a href="{0}">{1}</a></td> <td class="col2"><a href="{2}"><code>{3}</code></a><td> </tr>'.format(
-                    repo_link(repo_names[repo]), repo_names[repo], link(repo_names[repo], shas[repo]), short(shas[repo]), "" if first else 'class="separate"'))
+                separator = "" if first else 'class="separate"'
```
Comment (oliver-wm) [inline bamboo/summary.py:80]: could use enumerate here

**Review by oliver-wm — APPROVED** [review]

File: bamboo/summary.py (hunk 1 above)
Comment (lsf37) [reply bamboo/summary.py:107]: Good point, will update

File: bamboo/summary.py (hunk 2 above)
Comment (lsf37) [reply bamboo/summary.py]: Yes, after having tried it out, I think that is nicer.

**Pull request merged**